src/coreclr/jit/promotionliveness.cpp

```diff
@@ -0,0 +1,869 @@
+
+        if (m_compiler->compQmarkUsed)
+        {
+            for (Statement* stmt : block->Statements())
+            {
+                GenTree* dst;
+                GenTree* qmark = m_compiler->fgGetTopLevelQmark(stmt->GetRootNode(), &dst);
+                if (qmark == nullptr)
+                {
+                    for (GenTreeLclVarCommon* lcl : stmt->LocalsTreeList())
+                    {
+                        MarkUseDef(lcl, bb.VarUse, bb.VarDef);
+                    }
+                }
+                else
+                {
+                    for (GenTreeLclVarCommon* lcl : stmt->LocalsTreeList())
+                    {
+                        // Skip liveness updates/marking for defs; they may be conditionally executed.
+                        if ((lcl->gtFlags & GTF_VAR_DEF) == 0)
+                        {
+                            MarkUseDef(lcl, bb.VarUse, bb.VarDef);
+                        }
+                    }
+                }
+            }
+        }
+        else
+        {
+            for (Statement* stmt : block->Statements())
+            {
+                for (GenTreeLclVarCommon* lcl : stmt->LocalsTreeList())
+                {
+                    MarkUseDef(lcl, bb.VarUse, bb.VarDef);
+                }
+            }
+        }
+
+#ifdef DEBUG
+        if (m_compiler->verbose)
+        {
+            BitVec allVars(BitVecOps::Union(m_bvTraits, bb.VarUse, bb.VarDef));
+            printf(FMT_BB " USE(%u)=", block->bbNum, BitVecOps::Count(m_bvTraits, bb.VarUse));
+            DumpVarSet(bb.VarUse, allVars);
+            printf("\n" FMT_BB " DEF(%u)=", block->bbNum, BitVecOps::Count(m_bvTraits, bb.VarDef));
+            DumpVarSet(bb.VarDef, allVars);
+            printf("\n\n");
+        }
+#endif
+    }
+}
+
+//------------------------------------------------------------------------
+// MarkUseDef:
+//   Mark use/def information for a single appearence of a local.
+//
+// Parameters:
+//   lcl    - The local node
+//   useSet - The use set to mark in.
+//   defSet - The def set to mark in.
+//
+void PromotionLiveness::MarkUseDef(GenTreeLclVarCommon* lcl, BitSetShortLongRep& useSet, BitSetShortLongRep& defSet)
+{
+    AggregateInfo* agg = m_aggregates[lcl->GetLclNum()];
+    if (agg == nullptr)
+    {
+        return;
+    }
+
+    jitstd::vector<Replacement>& reps  = agg->Replacements;
+    bool                         isDef = (lcl->gtFlags & GTF_VAR_DEF) != 0;
+    bool                         isUse = !isDef;
+
+    unsigned  baseIndex  = m_structLclToTrackedIndex[lcl->GetLclNum()];
+    var_types accessType = lcl->TypeGet();
+
+    if (accessType == TYP_STRUCT)
+    {
+        if (lcl->OperIs(GT_LCL_ADDR))
+        {
+            // For LCL_ADDR this is a retbuf and we expect it to be a def. We
+            // don't know the exact size here so we cannot mark anything as
+            // being fully defined, thus we can just return.
+            assert(isDef);
```

Comment:
Yes, the problem is finding the call node that contains the retbuf, since the linked locals list does not contain it. In principle it shouldn't be hard to quickly find it utilizing `GTF_CALL`, so we could be a bit more precise about this.